### PR TITLE
User course input

### PR DESCRIPTION
**Pull request by aldy97**
# Changes
- This PR is following my previous one in the same branch.
- With addition to the previous changes, I added term and courseSection into redux.

## ScreenShots:
image
image
image
image

## Issues
- Since @johnagl's PR on the end point that takes all of term, department name and course number has not been merged yet, it can not be fully tested.
- But courseSections in redux can be properly updated using the current end point which only takes department name and course number. So I have a strong belief that it will work after the new end point is merged. 
- I should have updated seleted term to redux in home component which allows user to select which term, so currently the stored term in redux is always an empty string. I will do this in my next PR.

### CODE REVIEW

**Review by johnagl — APPROVED** [review]
Minor nitpick, for coursesAdded, can we add a space between coursedept and coursenumber? So CPSC 121 instead of CPSC121 :D. 

Other than that, awesome work! I'll approve this, and we can merge now, or merge when the endpoint is merged, up to you.

**Comment by aldy97** [comment]
@johnagl Thank you! I just modified rendering courses format in the list. I will wait until your PR is merged and see how mine works with your new end point :)
image

**Comment by johnagl** [comment]
The new endpoint is merged into main @aldy97 :)

**Pull request merged**
